R/db-postgres.r

```diff
@@ -1,3 +1,43 @@
+#' @export
+sql_select.PostgreSQLConnection <- function(con, select, from, where = NULL,
+                                       group_by = NULL, having = NULL,
+                                       order_by = NULL,
+                                       limit = NULL,
+                                       distinct = FALSE,
+                                       sample = NULL,
+                                       ...) {
+  out <- vector("list", 8)
+  names(out) <- c("select", "from", "where", "group_by", "having", "order_by",
+                  "limit", "sample")
+
+  out$select    <- sql_clause_select(select, con, distinct)
+  out$from      <- sql_clause_from(from, con)
+  out$where     <- sql_clause_where(where, con)
+  out$group_by  <- sql_clause_group_by(group_by, con)
+  out$having    <- sql_clause_having(having, con)
+  out$order_by  <- sql_clause_order_by(order_by, con)
+  out$limit     <- sql_clause_limit(limit, con)
+  size <- sample$size
```

Comment:
Rather than duplicating this big function, I think it would be better to have 

```R
sql_clause_sample <- fuction(sample, con) {
  UseMethod("sql_clause_sample", con)
}

sql_clause_sample.SQLiteConnection <- function(sample, con) {
   ...
}

```

SQLite would still need special treatment, but this would reduce the duplication significantly

Reply:
Does that make sense? Let me know if you need more details.

Reply:
Yes, it does. Thank you.